File: Segundo Período/Algoritmo e Estrutura de Dados - II/TP_01/Q11/isPalindromo.java
class isPalindromo {

    public static boolean isFim(String s) {
        if(s.length() == 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M') {
            return true;
        }

        return false;
    }

    public static boolean isPali(String s){ 
        return isPali(s,0);
    }
    
    public static boolean isPali(String s, int i){ 
        boolean resposta;

        if(i==s.length()){  

            resposta=true;

        }else if(s.charAt(i)!=s.charAt(s.length()-i-1)){ 

            resposta=false;
        }else{

            resposta=isPali(s,i+1); 
        }
        return resposta;
    }
    
    public static void main (String[] args){
        String entrada="";

        while(isFim(entrada)==false){

            entrada=MyIO.readLine();

            if(isPali(entrada)==true){
                
                System.out.println("SIM");

            }
            else if ((isPali(entrada)==false) && (isFim(entrada)==false)){ 
                System.out.println("NAO");

            }
        }
    }
}

File: Segundo Período/Algoritmo e Estrutura de Dados - II/TP_01/Q02/isPalindromo.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>


bool isFim(const char *s){
                                                                                                 
    if(strlen(s) == 3 && s[0] == 'F' && s[1] == 'I' && s[2] == 'M'){
        return true;
    }

        return false;
}
        
    

bool isPali(char *s){

    int cont = 0;
    bool result = false;

    for(int i = 0;i < strlen(s); i++){
        if(s[i] == s[strlen(s)-i-1]){
            cont++;
        }
    }
    if(cont==strlen(s)){
        result=true;
    }
    return result;
}


int main (){

    char entrada[1000];

    while (isFim(entrada)==false){

         scanf(" %[^\n]", entrada);

        if(isPali(entrada)==true){
               printf("SIM\n");

            }else if(isPali(entrada)==false){ //Segunda expressão foi necessária para que quando o usuário digite "FIM" a resposta nao seja exibida na tela 
                
                if(isFim(entrada)==true){
                    break;
                }

            printf("NAO\n");
        }
    }
}